### PR TITLE
git: Add delta.package option

### PR DESCRIPTION
**Pull request by sestrella**
### Description

Make it possible to customize the `delta` package

### Checklist

- [x] Change is backwards compatible.

- [x] Code formatted with `./format`.

- [ ] Code tested through `nix-shell --pure tests -A run.all`.

- [ ] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/docs/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [ ] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

  - [ ] Added myself and the module files to `.github/CODEOWNERS`.

### CODE REVIEW

File: modules/misc/news.nix
Comment (sumnerevans) [inline modules/misc/news.nix:1]: Small changes such as this do not need a news entry.

**Review by sumnerevans — CHANGES_REQUESTED** [review]
looks reasonable

File: modules/programs/git.nix [hunk 1]
```diff
@@ -310,6 +310,13 @@ in {
           '';
         };
 
+        package = mkOption {
+          type = types.package;
+          default = pkgs.delta;
+          defaultText = literalExpression "pkgs.delta";
+          description = "Package providing <command>delta</command>.";
+        };
```
Comment (ncfavier) [inline modules/programs/git.nix:318]:
```suggestion
        package = mkPackageOption pkgs "delta" { };
```

Comment (sestrella) [reply modules/programs/git.nix]: @ncfavier I refactored this section following your advice, thank you!

**Comment by rycee** [comment]
Thanks! Squashed and merged to master with a few minor edits in e21ec3db17c4b474f6bd9d8b967a3415fbc6b51d.

**Pull request closed without merging**